### PR TITLE
feat: add support for dataCy prop

### PR DESCRIPTION
**Pull request by LukeWlodarczyk**
Add support for dataCy prop

### CODE REVIEW

**Comment by ugogo** [comment]
Hey @LukeWlodarczyk, could you provide more information about why you need this prop?
It looks very specific

**Comment by LukeWlodarczyk** [comment]
I think it's quite common to use `data-cy` prop for testing using Cypress library

File: src/index.tsx [hunk 1]
```diff
@@ -209,6 +212,7 @@ const ReactInputVerificationCode = ({
               i === activeIndex ? 'is-active' : ''
             }`}
             onFocus={onItemFocus(i)}
+            data-cy={`${dataCy}-item`}
```
Comment (ugogo) [inline src/index.tsx:215]: wdyt about adding the index?

```suggestion
            data-cy={`${dataCy}-${i}-item`}
```

**Review by ugogo — COMMENTED** [review]
Minor comment otherwise LGTM 👍

File: src/index.tsx (hunk 1 above)
Comment (LukeWlodarczyk) [reply src/index.tsx]: Great idea 🚀

**Pull request merged**